datacommons/places.py

```diff
@@ -26,47 +26,113 @@
+
+    Raises:
+      ValueError: If the payload returned by the Data Commons REST API is
+        malformed.
+
+    Examples:
+      We would like to get all Counties contained in
+      `California <https://browser.datacommons.org/kg?dcid=geoId/06>`_. Specifying
+      the :code:`dcids` as a :obj:`list` result in the following.
+
+      >>> get_places_in(["geoId/06"], "County")
+      {
+        'geoId/06': [
+          'geoId/06041',
+          'geoId/06089',
+          'geoId/06015',
+          'geoId/06023',
+          'geoId/06067',
+          ...
+          # and 53 more
+        ]
+      }
+    """
+    dcids = list(dcids)
+    url = utils._API_ROOT + utils._API_ENDPOINTS['get_places_in']
+    payload = utils._send_request(url, req_json={
+        'dcids': dcids,
+        'place_type': place_type,
+    })
+
+    # Create the results and format it appropriately
+    result = utils._format_expand_payload(payload, 'place', must_exist=dcids)
+    return result
+
+
+def get_related_places(dcids, population_type, constraining_properties={},
+                       measured_prop='count',
+                       stat_type='measured', within_place='',
+                       per_capita=False, same_place_type=False):
+    """ Returns :obj:`Place`s related to :code:`dcids` for the given constraints.
+
+    Args:
+      dcids (:obj:`iterable` of :obj:`str`): Dcids to get contained in places.
+      population_type (:obj:`str`): The type of statistical population.
+      constraining_properties (:obj:`map` from :obj:`str` to :obj:`str`, optional):
+        A map from constraining property to the value that the
+        :obj:`StatisticalPopulation` should be constrained by.
+      measured_property (:obj:`str`): The measured property.
+      stat_type (:obj:`str`): The statistical type for the observation.
+      within_place(:obj:`str`): Optional, the place that all the related places
+        are contained in.
+      per_capita(:obj:`bool`): Optional, whether to take into account
+        `PerCapita` when compute the relatedness.
+      same_place_type(:obj:`bool`): Optional, whether to require all the
+        related places under the same ancestor place.
+
+    Returns:
+      The returned :obj:`Place`'s are formatted as a :obj:`dict` from a given
+      dcid to a list of related places for the given constraints.
+
+    Raises:
+      ValueError: If the payload returned by the Data Commons REST API is
+        malformed.
+
+    Examples:
+      We would like to get all related places of
+      `Santa Clara county <https://browser.datacommons.org/kg?dcid=geoId/06085>`
+      Specifying the :code:`dcids` as a :obj:`list` result in the following.
+
+      >>> get_places_in(["geoId/06"], "Person", {
+        "age": "Years21To64",
+        "gender": "Female"
+      }, "count", "measured")
+      {
+        'geoId/06085': [
+          'geoId/06041',
+          'geoId/06089',
+          'geoId/06015',
+          'geoId/06023',
+        ]
+      }
```

Comment:
Need to update example?

Reply:
The examples just show the format -- the result would change as we change algorithms to compute relatedness.